README.md

```diff
@@ -1095,6 +1128,192 @@ VTP file name. Valid values are a string or the keyword 'default'.
 ##### `password`
 Password for the VTP domain. Valid values are a string or the keyword 'default'.
 
+--
+### NetDev Resource Type Details
+
+The following resources are listed alphabetically.
+
+--
+
+### Type: domain_name
+
+Configure the domain name of the device
+
+#### Parameters
+
+##### `ensure`
+Determines whether or not the config should be present on the device. Valid values are 'present' and 'absent'. Default value is 'present'.
+
+##### `name`
+Domain name of the device. Valid value is a string.
+
+### Type: name_server
+
+#### Parameters
+
+##### `ensure`
+Determines whether or not the config should be present on the device. Valid values are 'present' and 'absent'. Default value is 'present'.
+
+##### `name`
+Hostname or address of the DNS server.  Valid value is a string.
+
+### Type: network_dns
+
+#### Parameters
+
+##### `ensure`
+Determines whether or not the config should be present on the device. Valid
+values are 'present' and 'absent'. Default value is 'present'.
+
+##### `name`
+Name, generally "settings", not used to manage the resource.  Valid value is a string.
+
+##### `domain`
+Default domain name to append to the device hostname.  Valid value is a string.
+
+##### `search`
+Array of DNS suffixes to search for FQDN entries.  Valid value is an array of strings.
+
+##### `servers`
+Array of DNS servers to use for name resolution.  Valid value is an array of strings.
+
+### Type: network_snmp
+
+#### Parameters
+
+##### `name`
+Name of the Puppet resource, not used to manage the device.  Valid value is a string.
+
+##### `enable`
+Enable or disable SNMP functionality.  Valid values are 'true' or 'false'.
+
+##### `contact`
+Contact name for this device.  Valid value is a string.
+
+##### `location`
+Location of this device.  Valid value is a string.
+
+### Type: ntp_config
+
+#### Parameters
+
+##### `ensure - Check this one.  Really ensureable?`
```

Comment:
All of these "Really ensureable" lines will be updated after I test them in the demo manifest.